packages/grafana-data/src/vector/FunctionalVector.ts

```diff
@@ -0,0 +1,73 @@
+import { vectorToArray } from './vectorToArray';
+import { Vector } from '../types';
+
+export abstract class FunctionalVector<T = any> implements Vector<T>, Iterable<T> {
+  abstract get length(): number;
+
+  abstract get(index: number): T;
+
+  // Implement "iterator protocol"
+  *iterator() {
+    for (let i = 0; i < this.length; i++) {
+      yield this.get(i);
+    }
+  }
+
+  // Implement "iterable protocol"
+  [Symbol.iterator]() {
+    return this.iterator();
```

Comment:
return vectorator(this)?  then remove the generator here.  seems like that should work

Reply:
:( i does not... all the other functins worked fine, but this one complains about type mismatch